### ISSUE TITLE
Debugging AL Language

### ISSUE THREAD

**Issue by FSchneiderYav** [microsoft/vscode#181485]
Type: <b>Bug</b>
I can't debug my AL project since the new update. The debugger is starting and I can make one step. Than the debugger is hanging on this line.

VS Code version: Code 1.78.0 (252e5463d60e63238250799aef7375787f68b4ee, 2023-05-03T20:09:00.748Z)
OS version: Windows_NT x64 10.0.22621
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|12th Gen Intel(R) Core(TM) i7-1265U (12 x 2688)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.69GB (0.95GB free)|
|Process Argv|--extensions-dir C:\\data\\VSCodeExtDev\\Extensions --user-data-dir C:\\data\\VSCodeExtDev\\UserData --crash-reporter-id 54bdd110-e8a9-44f0-9a67-7ddae2cda51d|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (2)</summary>

Extension|Author (truncated)|Version
---|---|---
csharp|ms-|1.25.7
al|ms-|11.0.787898

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30445986
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30410667
cppdebug:30492333
vsclangdc:30486549
c4g48928:30535728
dsvsc012:30540252
pynewext54:30695312
azure-dev_surveyone:30548225
3biah626:30602489
pyind779:30671433
f6dab269:30613381
pythonsymbol12:30671437
2i9eh265:30646982
6233i204:30730053
vsccsb:30705552
azdwalk:30721579
pythonms35:30701012
pythonfmttext:30731395
fixhidewlkth:30730051
hidesbindicator:30730055

```

</details>

**Comment by FSchneiderYav** [comment]
@AndriiBodashevskyi 
You can uninstall VS Code an install version 1.77.3
https://update.code.visualstudio.com/1.77.3/win32-x64/stable

**Comment by AndriiBodashevskyi** [comment]
@FSchneiderYav thank you, could you please tell me if I have settings sync on
![image](https://user-images.githubusercontent.com/61715396/236155037-820672ac-a107-43cb-8996-943a10eb4279.png)
will ALL my settings and extensions saved after I uninstall Visual Studio Code and install the 1.77.3 version? Thanks!

**Comment by FSchneiderYav** [comment]
The uninstalling doesn't delete the settings. Your settings and extensions are available after a new installation

**Comment by PeterConijn** [comment]
+1 

We are having this issue too since the last VSCode update. I am assuming a roll-back to the previous version will help until it automagically re-updates.

**Comment by DaanSchuitmaker** [comment]
Same issue here
Version: 1.78.0 (user setup)
Commit: 252e5463d60e63238250799aef7375787f68b4ee
Date: 2023-05-03T20:09:00.748Z
Electron: 22.4.8
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Windows_NT x64 10.0.22621
Sandboxed: No

**Comment by HegedusZsolt** [comment]
+1

**Comment by roblourens** [comment]
Please file an issue on the extension, thanks

**Comment by roblourens** [comment]
I got a report from another extension, I'll leave this open to track any changes needed on the VS Code side

**Comment by BewanBC** [comment]
Hello,
I would follow up this ticket because we encountered same issue
https://code.visualstudio.com/updates/v1_77 is working well. Seems the version of node.js debugger?

Thanks

**Comment by DeFynancy** [comment]
+1

**Comment by smorv** [comment]
I have the same issue. 
VS Code 1.78.0

**Comment by echaocat** [comment]
Hi I also have the same problem with the version 78 :(

**Comment by ekom21-TL** [comment]
Same here.

**Comment by horstoeko** [comment]
Hi,

same problem here:

- VSCode 1.78
- Windows 10 22H2
- ALLanguage 11.0 (Business Central 2023 Release Wave 1)

We urgently need a fix for this issue as we need VSCode for development in "Business Central 365"/ALLanguage. Currently we are not able to support our project team.

Regards

**Comment by FSchneiderYav** [comment]
> @horstoeko You can uninstall VS Code an install version 1.77.3 https://update.code.visualstudio.com/1.77.3/win32-x64/stable

**Comment by stefaanMLB** [comment]
I have the same issues woth Typescript/node projects

**Comment by minaskats** [comment]
+1 for Business Central , unable to debug.

**Comment by gjsjohnmurray** [comment]
Maybe #181825 is relevant

**Comment by dzw0218** [comment]
We use Python plugin and have same question.

**Comment by roblourens** [comment]
For anyone blocked by this, you can downgrade to 1.77 by downloading it from https://code.visualstudio.com/updates/v1_77. I'll get a fix out in 1.78.1 this week. Sorry for the trouble

**Comment by roblourens** [comment]
You can also use the VS Code Insiders build after tomorrow, which will have a fix.

I am reverting #181854 which will bring back another issue, but it's a much more minor issue. I will have to do a better fix for 1.79. 

Details- this code is clearly missing something: https://github.com/microsoft/vscode/blob/5967addf56035cd63c313ab2f3feaff4fcd11d66/src/vs/workbench/contrib/debug/common/debugModel.ts#L1313-L1336

It never resolves the `wholeCallStack` promise when called while `this.schedulers` is already initialized. This usually doesn't happen, but the AL debugger sends two Stopped events quickly for each step which leads to hitting this issue. It's probably possible to hit it in other more subtle ways too, so it will be nice to have this fix.

**Comment by roblourens** [comment]
Update- will be in _tomorrow's_ Insiders build (due to a security update, we weren't able to release the fix in today's).

And in 1.78.2, to be released either later this week or early next.

And again, you can downgrade to 1.77 to work around it as well. Thanks for your patience!

**Comment by christianbraeunlich** [comment]
> Has anyone else been able to verify that it works as expected in the Insiders build? That includes the change that will ship in 1.78.2

Heard from a colleague that it works as expected with the latest insiders build

**Comment by deadmouse** [comment]
For me it works fine 1.78.2 for AL Language

**Comment by Neoster** [comment]
I came across this issue too and started using the Insiders version (VSCode - Insiders V1.79.0). This is a doable workaround except for the fact that every F5 required me to authenticate my account using the popup and the code provided by VS Code.

_Note: Even though I've got Chrome setup as preferred browser in VS Code settings, the authentication page is show using my systems default browser (Safari). Running the non-insiders version of VS Code 1.78.2, it does respect the preferred browser setting in VS Code._

Today the debugger seems to be working fine in VS Code 1.78.2
